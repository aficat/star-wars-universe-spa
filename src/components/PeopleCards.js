import React, { Component } from 'react';
import PeopleCardsSingleCard from './PeopleCardsSingleCard';
import { fetchPeople } from '../actions/peopleActions';
import { connect } from 'react-redux';
import {CircularProgress, Typography} from '@material-ui/core';

// Renders Main Body of Star Wars People's Cards
class PeopleCards extends Component {

    // Fetch list of Star Wars People
    componentDidMount() {
        // activePage = 1; renders first page of people's results
        this.props.fetchPeople(1);
    }

    /**
     * @const  {Object} peopleCards - List of People Objects
     * 
     * @returns {} returns all PeopleCardsSingleCard component for each person
     */
    renderResult = () => {
        const peopleCards = this.props.people;
        return (
            <>
                {peopleCards && peopleCards.map(person =>
                    <PeopleCardsSingleCard
                        name={person.name}
                        gender={person.gender}
                        height={person.height}
                        mass={person.mass}
                        url={person.url}
                        eyeColor={person.eye_color}
                        hairColor={person.hair_color}
                        skinColor={person.skin_color}
                        birthYear={person.birth_year}
                        homeworldURL={person.homeworld}
                        filmsURLs={person.films}
                        speciesURLs={person.species}
                        vehiclesURLs={person.vehicles}
                        starshipsURLs={person.starships}
                        createdDate={person.created}
                        editedDate={person.edited}
                        key={person.url}
                    />
                )}
            </>
        )
    }

    render() {
        const { status, message } = this.props;
        switch (status) {
            case "done": // successfully retrieved people data
                return this.renderResult();
            case "retrieving": // retrieving people data
                return <div style={{margin: 50}}><CircularProgress/></div>;
            case "error": // error retrieving people data
                return (<div><Typography variant="body2" component="p" color="error">{message}</Typography></div>);
            default:
                return (<div><Typography variant="body2" component="p">{message}</Typography></div>);
        }
    }
}

const mapStateToProps = state => ({
    people: state.people.people,
    status: state.people.status,
    message: state.people.message
});
export default connect(mapStateToProps, { fetchPeople })(PeopleCards);
