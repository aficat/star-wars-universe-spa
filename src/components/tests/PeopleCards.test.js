import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import toJson from 'enzyme-to-json'
import PeopleCards from '../PeopleCards';
import store from '../../store';
import { Typography } from '@material-ui/core';

describe('PeopleCards component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<PeopleCards store={store} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders without crashing given the required props', () => {
        const props = {
            people: [],
            status: "retrieving",
            message: ""
        }
        const wrapper = shallow(<PeopleCards {...props} store={store} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

});