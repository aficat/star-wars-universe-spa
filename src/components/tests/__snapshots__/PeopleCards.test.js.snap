// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCards component renders without crashing given the required props 1`] = `
<ContextProvider
  value={null}
>
  <PeopleCards
    fetchPeople={[Function]}
    message="Retrieving people data..."
    people={Array []}
    status="retrieving"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
